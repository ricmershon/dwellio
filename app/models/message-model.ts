import { Document, Schema, Types, model, models } from 'mongoose';
import { MessageInputType } from '@/app/schemas/message-schema';

export interface MessageDocument extends MessageInputType, Document {
    sender: Types.ObjectId,
    recipient: Types.ObjectId,
    property: Types.ObjectId | { _id: Types.ObjectId, name: string },
    read: boolean,
    createdAt: Date;
    updatedAt: Date;
}

const MessageSchema = new Schema({
    sender: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    recipient: { type: Schema.Types.ObjectId, ref: 'User', required: true },
    property: { type: Schema.Types.ObjectId, ref: 'Property', required: true },
    name: { type: String, required: [true, 'Name is required'] },
    email: { type: String, required: [true, 'Email is required'] },
    phone: String,
    body: String,
    read: { type: Boolean, default: false }
}, {
    timestamps: true
});

const Message = models.Message || model<MessageDocument>('Message', MessageSchema);
export default Message;