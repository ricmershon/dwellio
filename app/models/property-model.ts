import { Document, Schema, Types, model, models } from 'mongoose';

import { PropertyImageData } from '@/app/lib/definitions';
import { PropertyInputType } from '@/app/schemas/property-schema';

export interface PropertyDocument extends Omit<PropertyInputType, 'imagesData'>, Document {
    owner: Types.ObjectId;
    isFeatured?: boolean;
    imagesData?: PropertyImageData[];
    createdAt: Date;
    updatedAt: Date;
}

const ImageSchema = new Schema({
    secureUrl: { type: String, required: true },
    publicId: { type: String, required: true }
}, { _id: false });

const PropertySchema = new Schema<PropertyDocument>({
    owner: { type: Schema.Types.ObjectId, required: true, ref: 'User' },
    name: { type: String, required: true },
    type: { type: String, required: true },
    description: { type: String  },
    location: {
        street: String,
        city: String,
        state: String,
        zipcode: String
    },
    beds: { type: Number, required: true },
    baths: { type: Number, required: true },
    squareFeet: { type: Number, required: true },
    amenities: [{ type: String }],
    rates: {
        nightly: Number,
        weekly: Number,
        monthly: Number
    },
    sellerInfo: {
        name: String,
        email: String,
        phone: String
    },
    imagesData: {
        type: [ImageSchema],
        required: true,
        validate: {
            validator: (value: PropertyImageData[]) => Array.isArray(value) && value.length > 0,
            message: 'At least one image is required in imagesData.',
        },
    },
    isFeatured: { type: Boolean, default: false }
}, {
    timestamps: true
});

const Property = models.Property || model<PropertyDocument>('Property', PropertySchema);
export default Property;