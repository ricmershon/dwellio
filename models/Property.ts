import { Document, Schema, Types, model, models } from 'mongoose';
import { UserInterface } from './User';

export interface PropertyInterface extends Document {
    _id: Types.ObjectId;
    owner: UserInterface;
    name: string;
    type: string;
    description?: string;
    location: {
        street: string;
        city: string;
        state: string;
        zipcode: string
    };
    beds: number;
    baths: number;
    square_feet: number;
    amenities?: Array<string>;
    rates: {
        nightly?: number;
        weekly?: number;
        monthly?: number
    };
    seller_info: {
        name: string;
        email: string;
        phone: string
    };
    images: Array<string>;
    is_featured: boolean;
    createdAt: Date;
    updatedAt: Date;
}

const PropertySchema = new Schema({
    _id: Schema.Types.ObjectId,
    owner: { type: Schema.Types.ObjectId, required: true, ref: 'User' },
    name: { type: String, required: true },
    type: { type: String, required: true },
    description: { type: String  },
    location: {
        street: String,
        city: String,
        state: String,
        zipcode: String
    },
    beds: { type: Number, required: true },
    baths: { type: Number, required: true },
    square_feet: { type: Number, required: true },
    amenities: [{ type: String }],
    rates: {
        nightly: Number,
        weekly: Number,
        monthly: Number
    },
    seller_info: {
        name: String,
        email: String,
        phone: String
    },
    images: [{ type: String }],
    is_featured: { type: Boolean, default: false }
}, {
    timestamps: true
});

// export const Property = models.Property || model('Property', PropertySchema);

export const Property = models.Property || model<PropertyInterface>('Property', PropertySchema);