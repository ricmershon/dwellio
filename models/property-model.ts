import { Document, Schema, Types, model, models } from "mongoose";

import { PropertyImageData } from "@/types/types";
import { PropertyInputType } from "@/schemas/property-schema";

export interface PropertyDocument extends Omit<PropertyInputType, "imagesData">, Document {
    owner: Types.ObjectId;
    isFeatured?: boolean;
    imagesData?: PropertyImageData[];
    createdAt: Date;
    updatedAt: Date;
}

const ImageSchema = new Schema({
    secureUrl: { type: String, required: true },
    publicId: { type: String, required: true },
    height: { type: Number, required: true },
    width: { type: Number, required: true },
}, { _id: false });

const PropertySchema = new Schema<PropertyDocument>({
    owner: { type: Schema.Types.ObjectId, required: true, ref: "User" },
    name: { type: String, required: true },
    type: { type: String, required: true },
    description: { type: String  },
    location: {
        street: String,
        city: String,
        state: String,
        zipcode: String
    },
    beds: { type: Number, required: true },
    baths: { type: Number, required: true },
    squareFeet: { type: Number, required: true },
    amenities: [{ type: String }],
    rates: {
        nightly: Number,
        weekly: Number,
        monthly: Number
    },
    sellerInfo: {
        name: String,
        email: String,
        phone: String
    },
    imagesData: {
        type: [ImageSchema],
        required: true,
        validate: {
            validator: (value: PropertyImageData[]) => Array.isArray(value) && value.length > 2,
            message: "At least three images are required.",
        },
    },
    isFeatured: { type: Boolean, default: false }
}, {
    timestamps: true
});

const Property = models.Property || model<PropertyDocument>("Property", PropertySchema);
export default Property;