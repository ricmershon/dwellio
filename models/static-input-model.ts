import { Schema, Document, model, models } from 'mongoose';

interface Amenity {
    id: string;
    value: string;
}

export interface StaticInputInterface extends Document {
    amenities: Amenity[];
    property_types: string[];
}

const AmenitySchema = new Schema<Amenity>({
    id: { type: String, required: true },
    value: { type: String, required: true }
}, {
    _id: false
});

const StaticInputSchema = new Schema<StaticInputInterface>({
    amenities: {
        type: [AmenitySchema],
        required: true
    },
    property_types: {
        type: [String],
        required: true
    }
});

const StaticInput = models.StaticInput || model<StaticInputInterface>('StaticInput', StaticInputSchema);
export default StaticInput;