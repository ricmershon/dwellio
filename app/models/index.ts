import Property, { PropertyInterface } from "@/app/models/property-model";
import User, { UserInterface} from "@/app/models/user-model";

export { Property, User };
export type { PropertyInterface, UserInterface }