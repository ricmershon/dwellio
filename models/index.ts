import Property, { PropertyDocument } from "@/models/property-model";
import User, { UserDocument} from "@/models/user-model";
import Message, { MessageDocument } from "@/models/message-model";
import StaticInputs, { StaticInputsDocument } from "@/models/static-inputs-model";

export { Property, User, Message, StaticInputs };
export type { PropertyDocument, UserDocument, MessageDocument, StaticInputsDocument }