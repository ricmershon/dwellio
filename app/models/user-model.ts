import { Document, Schema, Types, model, models } from 'mongoose';

import { PropertyDocument } from '@/app/models/property-document';

export interface UserDocument extends Document {
    username: string;
    email: string;
    image: string;
    favorites: Types.ObjectId[] & PropertyDocument[]
}

const UserSchema = new Schema({
    username: { type: String, required: [true, 'Username is required'], unique: true },
    email: {
        type: String,
        unique: [true, 'Email already exists'],
        required: [true, 'Email is required']
    },
    image: { type: String },
    favorites: [{ type: Schema.Types.ObjectId, ref: 'Property' }]
}, {
    timestamps: true
});

const User = models.User || model<UserDocument>('User', UserSchema);
export default User;