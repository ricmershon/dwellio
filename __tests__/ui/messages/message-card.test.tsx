import React from 'react';
import { render, screen } from '@/__tests__/test-utils';

import MessageCard from '@/ui/messages/message-card';
import { MessageDocument } from '@/models';

// Mock child components
jest.mock('@/ui/messages/toggle-message-read-button', () => {
    const MockToggleMessageReadButton = ({ messageId, read }: { messageId: string; read: boolean }) => (
        <div 
            data-testid="toggle-message-read-button"
            data-message-id={messageId}
            data-read={read}
        >
            {read ? 'Mark as Unread' : 'Mark as Read'}
        </div>
    );
    MockToggleMessageReadButton.displayName = 'MockToggleMessageReadButton';
    return MockToggleMessageReadButton;
});

jest.mock('@/ui/messages/delete-message-button', () => {
    const MockDeleteMessageButton = ({ messageId }: { messageId: string }) => (
        <div 
            data-testid="delete-message-button"
            data-message-id={messageId}
        >
            Delete
        </div>
    );
    MockDeleteMessageButton.displayName = 'MockDeleteMessageButton';
    return MockDeleteMessageButton;
});

describe('MessageCard', () => {
    const createMockMessage = (overrides = {}): MessageDocument => ({
        _id: 'message-123',
        sender: 'sender-123',
        recipient: 'recipient-123',
        property: {
            _id: 'property-123',
            name: 'Beautiful Downtown Condo'
        },
        name: 'John Doe',
        email: 'john.doe@example.com',
        phone: '+1-555-0123',
        body: 'I am interested in this property. Please contact me for more information.',
        read: false,
        createdAt: new Date('2024-01-15T10:30:00Z'),
        updatedAt: new Date('2024-01-15T10:30:00Z'),
        ...overrides
    } as unknown as MessageDocument);

    describe('Content Display', () => {
        it('should display property name from populated property object', () => {
            const message = createMockMessage();
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByText('Beautiful Downtown Condo')).toBeInTheDocument();
        });

        it('should render message body text', () => {
            const message = createMockMessage({
                body: 'Custom message content for testing purposes.'
            });
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByText('Custom message content for testing purposes.')).toBeInTheDocument();
        });

        it('should show formatted creation date with toLocaleString', () => {
            const message = createMockMessage({
                createdAt: new Date('2024-03-15T14:45:30Z')
            });
            
            render(<MessageCard message={message} />);
            
            const expectedDate = new Date('2024-03-15T14:45:30Z').toLocaleString();
            expect(screen.getByText(expectedDate)).toBeInTheDocument();
        });

        it('should display sender email with mailto link', () => {
            const message = createMockMessage({
                email: 'test.user@example.com'
            });
            
            render(<MessageCard message={message} />);
            
            const emailLink = screen.getByRole('link', { name: 'test.user@example.com' });
            expect(emailLink).toHaveAttribute('href', 'mailto:test.user@example.com');
            expect(emailLink).toHaveClass('text-blue-500');
        });

        it('should display sender phone with tel link', () => {
            const message = createMockMessage({
                phone: '+1-555-9999'
            });
            
            render(<MessageCard message={message} />);
            
            const phoneLink = screen.getByRole('link', { name: '+1-555-9999' });
            expect(phoneLink).toHaveAttribute('href', 'tel:+1-555-9999');
            expect(phoneLink).toHaveClass('text-blue-500');
        });
    });

    describe('Conditional Rendering', () => {
        it('should show "New" badge for unread messages', () => {
            const message = createMockMessage({ read: false });
            
            render(<MessageCard message={message} />);
            
            const newBadge = screen.getByText('New');
            expect(newBadge).toBeInTheDocument();
            expect(newBadge).toHaveClass('bg-yellow-500', 'text-white');
        });

        it('should hide "New" badge for read messages', () => {
            const message = createMockMessage({ read: true });
            
            render(<MessageCard message={message} />);
            
            expect(screen.queryByText('New')).not.toBeInTheDocument();
        });

        it('should handle property as object with name field', () => {
            const message = createMockMessage({
                property: {
                    _id: 'prop-456',
                    name: 'Luxury Apartment Complex'
                }
            });
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByText('Luxury Apartment Complex')).toBeInTheDocument();
        });

        it('should gracefully handle missing property data', () => {
            const message = createMockMessage({
                property: null
            });
            
            render(<MessageCard message={message} />);
            
            // Should not crash, property label should still be there
            expect(screen.getByText(/Property:/)).toBeInTheDocument();
        });

        it('should handle property without name field', () => {
            const message = createMockMessage({
                property: { _id: 'prop-789' }
            });
            
            render(<MessageCard message={message} />);
            
            // Should not crash, property label should still be there
            expect(screen.getByText(/Property:/)).toBeInTheDocument();
        });
    });

    describe('Link Generation', () => {
        it('should create correct mailto links with email addresses', () => {
            const message = createMockMessage({
                email: 'contact@realestate.com'
            });
            
            render(<MessageCard message={message} />);
            
            const emailLink = screen.getByRole('link', { name: 'contact@realestate.com' });
            expect(emailLink.getAttribute('href')).toBe('mailto:contact@realestate.com');
        });

        it('should create correct tel links with phone numbers', () => {
            const message = createMockMessage({
                phone: '(555) 123-4567'
            });
            
            render(<MessageCard message={message} />);
            
            const phoneLink = screen.getByRole('link', { name: '(555) 123-4567' });
            expect(phoneLink.getAttribute('href')).toBe('tel:(555) 123-4567');
        });

        it('should apply blue text color to contact links', () => {
            const message = createMockMessage();
            
            render(<MessageCard message={message} />);
            
            const emailLink = screen.getByRole('link', { name: message.email });
            const phoneLink = screen.getByRole('link', { name: message.phone });
            
            expect(emailLink).toHaveClass('text-blue-500');
            expect(phoneLink).toHaveClass('text-blue-500');
        });

        it('should handle special characters in email addresses', () => {
            const message = createMockMessage({
                email: 'user+test@example-domain.co.uk'
            });
            
            render(<MessageCard message={message} />);
            
            const emailLink = screen.getByRole('link', { name: 'user+test@example-domain.co.uk' });
            expect(emailLink.getAttribute('href')).toBe('mailto:user+test@example-domain.co.uk');
        });

        it('should handle special characters in phone numbers', () => {
            const message = createMockMessage({
                phone: '+44 (0)20 1234 5678'
            });
            
            render(<MessageCard message={message} />);
            
            const phoneLink = screen.getByRole('link', { name: '+44 (0)20 1234 5678' });
            expect(phoneLink.getAttribute('href')).toBe('tel:+44 (0)20 1234 5678');
        });
    });

    describe('Child Component Integration', () => {
        it('should render ToggleMessageReadButton with correct props', () => {
            const message = createMockMessage({ read: false });
            
            render(<MessageCard message={message} />);
            
            const toggleButton = screen.getByTestId('toggle-message-read-button');
            expect(toggleButton).toBeInTheDocument();
            expect(toggleButton).toHaveAttribute('data-message-id', 'message-123');
            expect(toggleButton).toHaveAttribute('data-read', 'false');
        });

        it('should render DeleteMessageButton with messageId', () => {
            const message = createMockMessage();
            
            render(<MessageCard message={message} />);
            
            const deleteButton = screen.getByTestId('delete-message-button');
            expect(deleteButton).toBeInTheDocument();
            expect(deleteButton).toHaveAttribute('data-message-id', 'message-123');
        });

        it('should pass message read state to toggle button', () => {
            const readMessage = createMockMessage({ read: true });
            
            render(<MessageCard message={readMessage} />);
            
            const toggleButton = screen.getByTestId('toggle-message-read-button');
            expect(toggleButton).toHaveAttribute('data-read', 'true');
            expect(toggleButton).toHaveTextContent('Mark as Unread');
        });

        it('should maintain component layout and spacing', () => {
            const message = createMockMessage();
            const { container } = render(<MessageCard message={message} />);
            
            // Check that both buttons are present in the card
            expect(screen.getByTestId('toggle-message-read-button')).toBeInTheDocument();
            expect(screen.getByTestId('delete-message-button')).toBeInTheDocument();
            
            // Verify card container structure
            const card = container.firstChild as HTMLElement;
            expect(card).toHaveClass('relative', 'bg-white', 'p-4', 'rounded-md');
        });
    });

    describe('Data Handling', () => {
        it('should handle MessageDocument interface correctly', () => {
            const message = createMockMessage();
            
            expect(() => render(<MessageCard message={message} />)).not.toThrow();
        });

        it('should type-cast _id as string for child components', () => {
            const message = createMockMessage({
                _id: 'test-id-string'
            });
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByTestId('toggle-message-read-button')).toHaveAttribute('data-message-id', 'test-id-string');
            expect(screen.getByTestId('delete-message-button')).toHaveAttribute('data-message-id', 'test-id-string');
        });

        it('should access nested property object safely', () => {
            const message = createMockMessage({
                property: {
                    _id: 'nested-prop',
                    name: 'Nested Property Name'
                }
            });
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByText('Nested Property Name')).toBeInTheDocument();
        });

        it('should format dates consistently', () => {
            const testDate = new Date('2024-06-01T08:15:30Z');
            const message = createMockMessage({
                createdAt: testDate
            });
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByText(testDate.toLocaleString())).toBeInTheDocument();
        });
    });

    describe('Styling and Layout', () => {
        it('should apply card styling with shadow, border, and padding', () => {
            const message = createMockMessage();
            const { container } = render(<MessageCard message={message} />);
            
            const card = container.firstChild as HTMLElement;
            expect(card).toHaveClass(
                'relative',
                'bg-white',
                'p-4',
                'rounded-md',
                'text-sm',
                'shadow-md',
                'border',
                'border-gray-100',
                'text-gray-700'
            );
        });

        it('should position "New" badge in top-right corner', () => {
            const message = createMockMessage({ read: false });
            
            render(<MessageCard message={message} />);
            
            const newBadge = screen.getByText('New');
            expect(newBadge).toHaveClass(
                'absolute',
                'top-3',
                'right-3',
                'bg-yellow-500',
                'text-white',
                'px-2',
                'py-1',
                'text-sm',
                'rounded-md'
            );
        });

        it('should maintain proper text hierarchy and spacing', () => {
            const message = createMockMessage();
            
            render(<MessageCard message={message} />);
            
            // Check heading has bottom margin
            const heading = screen.getByRole('heading', { level: 2 });
            expect(heading).toHaveClass('mb-4');
            
            // Check list has top margin
            const list = screen.getByRole('list');
            expect(list).toHaveClass('mt-4');
        });

        it('should handle long message content gracefully', () => {
            const longMessage = 'A'.repeat(1000);
            const message = createMockMessage({
                body: longMessage
            });
            
            render(<MessageCard message={message} />);
            
            expect(screen.getByText(longMessage)).toBeInTheDocument();
        });

        it('should render as a div with proper semantic structure', () => {
            const message = createMockMessage();
            const { container } = render(<MessageCard message={message} />);
            
            const card = container.firstChild as HTMLElement;
            expect(card.tagName).toBe('DIV');
            
            // Check semantic elements exist
            expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
            expect(screen.getByRole('list')).toBeInTheDocument();
        });
    });

    describe('Edge Cases', () => {
        it('should handle empty message body', () => {
            const message = createMockMessage({ body: '' });
            
            render(<MessageCard message={message} />);
            
            // Should still render the card structure
            expect(screen.getByRole('heading')).toBeInTheDocument();
        });

        it('should handle missing email', () => {
            const message = createMockMessage({ email: '' });
            
            render(<MessageCard message={message} />);
            
            // Should still render email label but with empty link
            expect(screen.getByText(/Reply Email:/)).toBeInTheDocument();
        });

        it('should handle missing phone', () => {
            const message = createMockMessage({ phone: '' });
            
            render(<MessageCard message={message} />);
            
            // Should still render phone label but with empty link
            expect(screen.getByText(/Reply Phone:/)).toBeInTheDocument();
        });

        it('should handle different date formats', () => {
            const message = createMockMessage({
                createdAt: new Date('2023-12-25')
            });
            
            render(<MessageCard message={message} />);
            
            const expectedDate = new Date('2023-12-25').toLocaleString();
            expect(screen.getByText(expectedDate)).toBeInTheDocument();
        });
    });

    describe('Snapshots', () => {
        it('should match snapshot for read message state', () => {
            const message = createMockMessage({ read: true });
            const { container } = render(<MessageCard message={message} />);
            expect(container.firstChild).toMatchSnapshot('read-message-state');
        });

        it('should match snapshot for unread message state (with "New" badge)', () => {
            const message = createMockMessage({ read: false });
            const { container } = render(<MessageCard message={message} />);
            expect(container.firstChild).toMatchSnapshot('unread-message-with-new-badge');
        });

        it('should match snapshot for long message content', () => {
            const longMessage = createMockMessage({
                body: 'This is a very long message that contains multiple sentences and should test how the component handles extensive text content. It includes various details about the property inquiry and demonstrates the component\'s ability to display lengthy communications from potential tenants or property owners. The message continues with more information about specific requirements, preferences, and questions about the property features, amenities, and availability.'
            });
            const { container } = render(<MessageCard message={longMessage} />);
            expect(container.firstChild).toMatchSnapshot('long-message-content');
        });

        it('should match snapshot for different property types', () => {
            const apartmentMessage = createMockMessage({
                property: { name: 'Luxury Downtown Apartment Complex' }
            });
            const { container } = render(<MessageCard message={apartmentMessage} />);
            expect(container.firstChild).toMatchSnapshot('apartment-property-type');
        });

        it('should match snapshot for house property type', () => {
            const houseMessage = createMockMessage({
                property: { name: 'Spacious Family House with Garden' }
            });
            const { container } = render(<MessageCard message={houseMessage} />);
            expect(container.firstChild).toMatchSnapshot('house-property-type');
        });

        it('should match snapshot with all contact information', () => {
            const fullContactMessage = createMockMessage({
                email: 'john.doe@example.com',
                phone: '555-123-4567',
                body: 'Complete message with all contact details'
            });
            const { container } = render(<MessageCard message={fullContactMessage} />);
            expect(container.firstChild).toMatchSnapshot('full-contact-information');
        });

        it('should match snapshot with minimal contact information', () => {
            const minimalMessage = createMockMessage({
                email: '',
                phone: '',
                body: 'Message with minimal contact info'
            });
            const { container } = render(<MessageCard message={minimalMessage} />);
            expect(container.firstChild).toMatchSnapshot('minimal-contact-information');
        });
    });
});