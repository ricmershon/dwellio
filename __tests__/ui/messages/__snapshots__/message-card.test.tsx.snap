// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`MessageCard Snapshots should match snapshot for different property types: apartment-property-type 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <div
    class="absolute top-3 right-3 bg-yellow-500 text-white px-2 py-1 text-sm rounded-md"
  >
    New
  </div>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Luxury Downtown Apartment Complex
  </h2>
  <p>
    I am interested in this property. Please contact me for more information.
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:john.doe@example.com"
      >
        john.doe@example.com
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:+1-555-0123"
      >
        +1-555-0123
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="false"
    data-testid="toggle-message-read-button"
  >
    Mark as Read
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;

exports[`MessageCard Snapshots should match snapshot for house property type: house-property-type 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <div
    class="absolute top-3 right-3 bg-yellow-500 text-white px-2 py-1 text-sm rounded-md"
  >
    New
  </div>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Spacious Family House with Garden
  </h2>
  <p>
    I am interested in this property. Please contact me for more information.
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:john.doe@example.com"
      >
        john.doe@example.com
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:+1-555-0123"
      >
        +1-555-0123
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="false"
    data-testid="toggle-message-read-button"
  >
    Mark as Read
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;

exports[`MessageCard Snapshots should match snapshot for long message content: long-message-content 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <div
    class="absolute top-3 right-3 bg-yellow-500 text-white px-2 py-1 text-sm rounded-md"
  >
    New
  </div>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Beautiful Downtown Condo
  </h2>
  <p>
    This is a very long message that contains multiple sentences and should test how the component handles extensive text content. It includes various details about the property inquiry and demonstrates the component's ability to display lengthy communications from potential tenants or property owners. The message continues with more information about specific requirements, preferences, and questions about the property features, amenities, and availability.
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:john.doe@example.com"
      >
        john.doe@example.com
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:+1-555-0123"
      >
        +1-555-0123
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="false"
    data-testid="toggle-message-read-button"
  >
    Mark as Read
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;

exports[`MessageCard Snapshots should match snapshot for read message state: read-message-state 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Beautiful Downtown Condo
  </h2>
  <p>
    I am interested in this property. Please contact me for more information.
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:john.doe@example.com"
      >
        john.doe@example.com
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:+1-555-0123"
      >
        +1-555-0123
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="true"
    data-testid="toggle-message-read-button"
  >
    Mark as Unread
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;

exports[`MessageCard Snapshots should match snapshot for unread message state (with "New" badge): unread-message-with-new-badge 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <div
    class="absolute top-3 right-3 bg-yellow-500 text-white px-2 py-1 text-sm rounded-md"
  >
    New
  </div>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Beautiful Downtown Condo
  </h2>
  <p>
    I am interested in this property. Please contact me for more information.
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:john.doe@example.com"
      >
        john.doe@example.com
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:+1-555-0123"
      >
        +1-555-0123
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="false"
    data-testid="toggle-message-read-button"
  >
    Mark as Read
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;

exports[`MessageCard Snapshots should match snapshot with all contact information: full-contact-information 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <div
    class="absolute top-3 right-3 bg-yellow-500 text-white px-2 py-1 text-sm rounded-md"
  >
    New
  </div>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Beautiful Downtown Condo
  </h2>
  <p>
    Complete message with all contact details
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:john.doe@example.com"
      >
        john.doe@example.com
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:555-123-4567"
      >
        555-123-4567
      </a>
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="false"
    data-testid="toggle-message-read-button"
  >
    Mark as Read
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;

exports[`MessageCard Snapshots should match snapshot with minimal contact information: minimal-contact-information 1`] = `
<div
  class="relative bg-white p-4 rounded-md text-sm shadow-md border border-gray-100 text-gray-700"
>
  <div
    class="absolute top-3 right-3 bg-yellow-500 text-white px-2 py-1 text-sm rounded-md"
  >
    New
  </div>
  <h2
    class="mb-4"
  >
    <span
      class="font-bold"
    >
      Property:
    </span>
     
    Beautiful Downtown Condo
  </h2>
  <p>
    Message with minimal contact info
  </p>
  <ul
    class="mt-4"
  >
    <li>
      <span
        class="font-bold"
      >
        Reply Email: 
      </span>
      <a
        class="text-blue-500"
        href="mailto:"
      />
    </li>
    <li>
      <span
        class="font-bold"
      >
        Reply Phone: 
      </span>
      <a
        class="text-blue-500"
        href="tel:"
      />
    </li>
    <li>
      <span
        class="font-bold"
      >
        Received: 
      </span>
      1/15/2024, 5:30:00 AM
    </li>
  </ul>
  <div
    data-message-id="message-123"
    data-read="false"
    data-testid="toggle-message-read-button"
  >
    Mark as Read
  </div>
  <div
    data-message-id="message-123"
    data-testid="delete-message-button"
  >
    Delete
  </div>
</div>
`;
